music21/harmony.py

```diff
@@ -1809,27 +1837,19 @@ def _parseFigure(self):
             bass = bass.replace('/', '')
             self.bass(bass)
             #remove the root and bass from the string and any additions/omitions/alterations/
-            remaining = st.replace(m2.group(), '')
+            # get the match on st, should match since st is a substring of
+            # prelimFigure
+            m3 = re.search(r'/[A-Ga-g][#-]*', st)
+            assert m3
```

Comment:
bare asserts not allowed in music21 code base.  Does not tell the user what to do to fix.  Raise an error with a message that can be trapped.

Reply:
I want to comment on this: why are asserts not allowed? I agree it is completely wrong to use asserts to convey error messages to users. They are useful however for debugging purposes, in order to formally specify invariants that should always be satisfied in any execution, i.e. regardless of user "mistakes". In that sense, they serve the same purpose as unit tests. In fact, running a piece of code and verifying that all asserts pass is a valuable unit test in itself.

Here, I'm just saying that 'm3' should always match since 'st' is a substring of 'prelimFigure'. I could just write it as a comment, but I might as well check it. If this assert fails, it's a programming error of mine, and not a user error, so this should never be reported to the user, but rather, the code should be sufficiently tested to ensure that the assert never fails.

Anyway, this might boil down to coding style, so I'm happy to simply remove this line. Furthermore, Python has the nasty behaviour of enabling asserts by default (i.e. having '__debug__' on by default), when the opposite should be the default, so that the asserts are only verified at testing phase, and this invalidates my argument to some extent.